### PR TITLE
SBT: treat warnings as errors

### PR DESCRIPTION
**Pull request by kitbellew**

### CODE REVIEW

**Comment by tgodzik** [comment]
> and this comes from the pprint copy, looks like the previous case supersedes the match on TypeBounds:

We could ignore those via Wconf like in metals:
```
....
    "-Wconf:src=*.MtagsIndexer.scala&msg=parameter owner in method visitOccurrence:silent",
    "-Wconf:src=*.ScaladocParser.scala&msg=parameter (pos|message) in method reportError:silent",
```
https://github.com/scalameta/metals/blob/main/build.sbt#L154


I am digging into the other issue.

**Comment by tgodzik** [comment]
Ach, wait, I thought it's scalameta/scalameta repo :sweat_smile: 

Looks like TypeRepr is a type alias for Type, which is why that is failing. We should switch the cases around

**Comment by tgodzik** [comment]
The other issue is that mdoc depends on metaconfig and metaconfig-docs depend on metaconfig also obviously. But that module is just for documentation, so we can safely turn of that warning via exclude or just ignoring it. Do you want me to finish this PR?

**Comment by kitbellew** [comment]
> Ach, wait, I thought it's scalameta/scalameta repo 😅
> 
> Looks like TypeRepr is a type alias for Type, which is why that is failing. We should switch the cases around

i thought so, too, but then it looked like you simply copied this code from somewhere... was it fixed there? maybe the fix is not to switch them around but to remove the second case?

**Comment by kitbellew** [comment]
> The other issue is that mdoc depends on metaconfig and metaconfig-docs depend on metaconfig also obviously. But that module is just for documentation, so we can safely turn of that warning via exclude or just ignoring it. Do you want me to finish this PR?

Yes, please!!!

File: metaconfig-tests/shared/src/test/scala/metaconfig/ConfDynamicSuite.scala [hunk 1]
```diff
@@ -1,4 +1,6 @@
 package metaconfig
+import metaconfig.Configured.NotOk
+import metaconfig.Configured.Ok
```
Comment (kitbellew) [inline metaconfig-tests/shared/src/test/scala/metaconfig/ConfDynamicSuite.scala:3]: minor, feel free to skip all of this comment:

- i'd add a blank line after package
- i'd import as a wildcard `metaconfig.Configured._`

or i'd simply use `case Configured.Xxx(...)` below, like they did before, without adding imports.

Comment (tgodzik) [reply metaconfig-tests/shared/src/test/scala/metaconfig/ConfDynamicSuite.scala]: done

**Comment by kitbellew** [comment]
thank you for helping with this.

one practical question: how did you manage to push changes into my fork, to continue with this pr?

**Comment by tgodzik** [comment]
> thank you for helping with this.
> 
> one practical question: how did you manage to push changes into my fork, to continue with this pr?

I think that' a default option to allow contributors to push to the specific branch.

**Pull request merged**